src/libraries/Microsoft.Win32.SystemEvents/src/Microsoft/Win32/SystemEvents.cs

```diff
@@ -77,32 +77,26 @@ private static unsafe bool UserInteractive
         {
             get
             {
-                if (Environment.OSVersion.Platform == System.PlatformID.Win32NT)
-                {
-                    IntPtr hwinsta = IntPtr.Zero;
+                IntPtr hwinsta = IntPtr.Zero;
 
-                    hwinsta = Interop.User32.GetProcessWindowStation();
-                    if (hwinsta != IntPtr.Zero && s_processWinStation != hwinsta)
-                    {
-                        s_isUserInteractive = true;
+                hwinsta = Interop.User32.GetProcessWindowStation();
+                if (hwinsta != IntPtr.Zero && s_processWinStation != hwinsta)
+                {
+                    s_isUserInteractive = true;
 
-                        int lengthNeeded = 0;
-                        Interop.User32.USEROBJECTFLAGS flags = default;
+                    int lengthNeeded = 0;
+                    Interop.User32.USEROBJECTFLAGS flags = default;
 
-                        if (Interop.User32.GetUserObjectInformationW(hwinsta, Interop.User32.UOI_FLAGS, ref flags, sizeof(Interop.User32.USEROBJECTFLAGS), ref lengthNeeded))
+                    if (Interop.User32.GetUserObjectInformationW(hwinsta, Interop.User32.UOI_FLAGS, ref flags, sizeof(Interop.User32.USEROBJECTFLAGS), ref lengthNeeded))
```

Comment:
The PInvoke definition of GetUserObjectInformation is not quite right:
- The 3rd argument should better be `void*` so that it can be used to any type of information and not just this specific one (`void*` has smaller marshaling overhead as well)
- The 4th and 5th arguments should be `uint`

Reply:
ah yes, I didn't see it was PVOID and that is called out. https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/interop-pinvokes.md